server/rest/src/main/java/org/infinispan/rest/resources/ProtobufResource.java

```diff
@@ -0,0 +1,233 @@
+      return builder.getHttpStatus() == HttpResponseStatus.CREATED || builder.getHttpStatus() == HttpResponseStatus.OK;
+   }
+
+   private CompletionStage<RestResponse> getSchema(RestRequest request) {
+      String schemaName = checkMandatorySchemaName(request);
+
+      AdvancedCache<Object, Object> cache = invocationHelper.getRestCacheManager()
+            .getCache(ProtobufMetadataManager.PROTOBUF_METADATA_CACHE_NAME, request);
+
+      RestCacheManager<Object> restCacheManager = invocationHelper.getRestCacheManager();
+      return restCacheManager.getPrivilegedInternalEntry(cache, schemaName, true).thenApply(entry -> {
+         NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+         if(entry == null) {
+            responseBuilder.status(HttpResponseStatus.NOT_FOUND.code());
+         } else {
+            responseBuilder.status(HttpResponseStatus.OK.code());
+            responseBuilder.contentType(MediaType.TEXT_PLAIN);
+            responseBuilder.entity(entry.getValue());
+         }
+         return responseBuilder.build();
+      });
+   }
+
+   private CompletionStage<RestResponse> deleteSchema(RestRequest request) {
+      String schemaName = checkMandatorySchemaName(request);
+
+      RestCacheManager<Object> restCacheManager = invocationHelper.getRestCacheManager();
+      AdvancedCache<Object, Object> protobufCache = restCacheManager.getCache(ProtobufMetadataManager.PROTOBUF_METADATA_CACHE_NAME, request);
+
+      return restCacheManager.getPrivilegedInternalEntry(protobufCache, schemaName, true).thenCompose(entry -> {
+         NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+         responseBuilder.status(HttpResponseStatus.NOT_FOUND);
+
+         if (entry instanceof InternalCacheEntry) {
+               responseBuilder.status(HttpResponseStatus.NO_CONTENT.code());
+               return restCacheManager.remove(ProtobufMetadataManager.PROTOBUF_METADATA_CACHE_NAME, schemaName,
+                     MediaType.MATCH_ALL, request)
+                     .thenApply(v -> responseBuilder.build());
+         }
+         return CompletableFuture.completedFuture(responseBuilder.build());
+      });
+   }
+
+   private ValidationError createErrorContent(String schemaName, String cause) {
+      String message = "Schema " + schemaName + " has errors";
+      ValidationError validationError = new ValidationError();
+      validationError.message = message;
+      validationError.cause = cause;
+      return validationError;
+   }
+
+   private String checkMandatorySchemaName(RestRequest request) {
+      String schemaName = request.variables().get("schemaName");
+      if (schemaName == null)
+         throw new NoKeyException("schemaName");
+      return schemaName.endsWith(ProtobufMetadataManager.PROTO_KEY_SUFFIX) ? schemaName : schemaName + ProtobufMetadataManager.PROTO_KEY_SUFFIX;
+   }
+
+   static class ValidationError {
```

Comment:
Unfortunately this isn't quite a record class. Please make a constructor with final fields.

Reply:
Or does Jackson not like that?

Reply:
all the other classes are done like this and I recall it is related